src/SqlAsyncCollector.cs

```diff
@@ -313,6 +313,10 @@ private static void GenerateDataQueryForMerge(TableInformation table, IEnumerabl
                 else
                 {
                     // ToDo: add check for duplicate primary keys once we find a way to get primary keys.
+                    if (table.Comparer == StringComparer.OrdinalIgnoreCase)
+                    {
+                        ChangePropertiesToLowerCase(JObject.Parse(row.ToString()));
```

Comment:
How is this actually changing anything? You convert the JObject to a string and then turn it back into a JObject which you then modify. But the original row isn't modified at all...